### PR TITLE
REFAC(overlay,plugins): replace NULL with nullptr

### PR DESCRIPTION
**Pull request by Popkornium18**
This changes all occurances of NULL in the plugins and overlay source dir to nullptr. Additionally explicit comparisons with NULL were removed.

Implements #3763 (partly)

### CODE REVIEW

File: overlay/olsettings.cpp [hunk 1]
```diff
@@ -58,17 +58,17 @@ static std::vector<std::string> regReadMultiString(HKEY key,
 {
 	LONG err = 0;
 	std::vector<std::string> out;
-	char *buf = NULL;
+	char *buf = nullptr;
 	HKEY subKeyHandle = 0;
 
-	err = RegOpenKeyExA(key, subKey.c_str(), NULL, KEY_READ, &subKeyHandle);
+	err = RegOpenKeyExA(key, subKey.c_str(), nullptr, KEY_READ, &subKeyHandle);
```
Comment (Popkornium18) [inline overlay/olsettings.cpp:64]: So if I understand this correctly this NULL should have been a 0 in the first place?
https://docs.microsoft.com/en-us/windows/win32/api/winreg/nf-winreg-regopenkeyexa

Sorry if that's a dumb question but I have no win32 experience and no Windows compiler at hand.

Comment (Krzmbrzl) [reply overlay/olsettings.cpp:64]: I think so, yes :thinking:

**Comment by Krzmbrzl** [comment]
The only part missing is the server-code now, right? :)

**Comment by Popkornium18** [comment]
Yes, I'll do that soon.

**Comment by Krzmbrzl** [comment]
Nice! - thanks :+1:

**Pull request merged**